Add markdown Readme for System.Text.Json (#69015)
dotnet/runtime: src/libraries/System.Text.Json/src/README.md

## About

Provides high-performance and low-allocating types that serialize objects to JavaScript Object Notation (JSON) text and deserialize JSON text to objects, with UTF-8 support built-in. Also provides types to read and write JSON text encoded as UTF-8, and to create an in-memory document object model (DOM), that is read-only, for random access of the JSON elements within a structured view of the data.

> The `System.Text.Json` library is built-in as part of the shared framework for .NET Core 3.0+ and .NET 5+. You only need to install it as a package for earlier .NET Core versions or other target frameworks.

Reviewer: ```suggestion
The `System.Text.Json` library is built-in as part of the shared framework for .NET Core 3.0+ and .NET 5+. The package can be installed when needing to access more recent versions of the library in older target frameworks.
```

> The `System.Text.Json` library is built-in as part of the shared framework for .NET Core 3.0+ and .NET 5+. The package can be installed when needing to access more recent versions of the library in older target frameworks.

Reviewer 2: ```suggestion
The `System.Text.Json` library is built-in as part of the shared framework for .NET 5+. The package can be installed when needing to access more recent versions of the library in older target frameworks.
```
.NET Core 3.0 is going to be EOL when this package ships. No need to mention in the docs.

Reviewer 3: .NET 5 is also EOL. The minimum supported versions that we support are the ones that we build this library for: .NET 6, .NET Standard 2.0 and .NET Framework 4.6.2

Author: It was meant not to document target frameworks, but to explain when users need to install a package. It does not have to mention exact versions if that's problematic. How about:

>The System.Text.Json library is built-in as part of the shared framework in .NET Runtime. The package can be installed when you need to use it in other target frameworks.

(implyilng that other target frameworks are supported versions of .NET Standard and .NET Framework)

Reviewer: Yes, that sounds appropriate and likely not subject to change.
